### PR TITLE
Feat: CoJ sync

### PR DESCRIPTION
**Pull request by ReubenRobertsHEE**
One possible issue with this implementation is that the `receivedFromTis` timestamp will be updated every time the record is received from DMS, so if we redrive the table all the entries will be replaced. This means it will cease to be a reliable source of data to estimate turnaround time. I think maybe I'll adjust to persist the first value it was set to.

Edit: this has been fixed in the latest commit.

TIS21-4717

### CODE REVIEW

File: src/main/java/uk/nhs/hee/tis/trainee/sync/dto/ConditionsOfJoiningDto.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+ * A conditions of joining DTO for inclusion in a Programme Membership.
+ */
+@Data
+public class ConditionsOfJoiningDto {
+
+  private Instant signedAt;
+  private String version;
+  private Instant receivedFromTis;
```
Comment (ReubenRobertsHEE) [inline src/main/java/uk/nhs/hee/tis/trainee/sync/dto/ConditionsOfJoiningDto.java:35]: I don't particularly love this name, if there are any better ideas. Maybe 'confirmedSavedAt' or 'receiptConfirmedAt'?

**Comment by ReubenRobertsHEE** [comment]
Note that a SQS conditions-of-joining queue was created but actually isn't needed, hence the lack of any config or task-definition-template entry.

File: src/main/java/uk/nhs/hee/tis/trainee/sync/dto/ConditionsOfJoiningDto.java (hunk 1 above)
Comment (EdwardBarclay) [reply src/main/java/uk/nhs/hee/tis/trainee/sync/dto/ConditionsOfJoiningDto.java:35]: the only alternatives I can think of are 'synchedAt' or 'syncConfirmedAt'

**Review by EdwardBarclay — APPROVED** [review]

File: src/main/java/uk/nhs/hee/tis/trainee/sync/dto/ConditionsOfJoiningDto.java (hunk 1 above)
Comment (ReubenRobertsHEE) [reply src/main/java/uk/nhs/hee/tis/trainee/sync/dto/ConditionsOfJoiningDto.java:35]: I like 'synchedAt' - it pairs up nicely with signedAt. Let me make the change...

**Review by EdwardBarclay — APPROVED** [review]

**Pull request merged**